cmd/gdeploy/main.go

```diff
@@ -67,7 +67,7 @@ func main() {
 			WithBeforeFuncs(&sso.SSOCommand, RequireDeploymentConfig(), VerifyGDeployCompatibility(), RequireAWSCredentials()),
 			WithBeforeFuncs(&backup.Command, RequireDeploymentConfig(), PreventDevUsage(), VerifyGDeployCompatibility(), RequireAWSCredentials()),
 			WithBeforeFuncs(&restore.Command, RequireDeploymentConfig(), PreventDevUsage(), VerifyGDeployCompatibility(), RequireAWSCredentials()),
-			WithBeforeFuncs(&provider.Command, RequireDeploymentConfig(), VerifyGDeployCompatibility(), RequireAWSCredentials()),
+			WithBeforeFuncs(&provider.Command, RequireDeploymentConfig(), VerifyGDeployCompatibility()),
```

Comment:
We need to make sure that we still have consistent error messages when the user choosed the CLI based flow.

We could add the call to RequireAWSCredentials in the "add" command itself when the if block for prompt CLI is entered.

Reply:
Good call here, the reason I removed this from the `WithBeforeFuncs` stack was because it wasn't needed when using the UI based flow. I'll add it to the command itself when the CLI flow is used.